optashift-employee-rostering-server/src/main/java/org/optaplanner/openshift/employeerostering/server/roster/RosterGenerator.java

```diff
@@ -432,6 +433,44 @@ private Integer createTenant(int spotListSize, int employeeListSize, int lengthI
         return employeeList;
     }
 
+    private List<EmployeeAvailability> createEmployeeAvailabilityList(int tenantId, TenantConfiguration config, List<Employee> employeeList, LocalDate fromDate, LocalDate toDate) {
+        List<EmployeeAvailability> out = new ArrayList<>();
+        List<LocalDate> datesBetween = new ArrayList<>();
+        for (LocalDate currDate = fromDate; currDate.isBefore(toDate); currDate = currDate.plusDays(1)) {
+            datesBetween.add(currDate);
+        }
+        for (LocalDate date : datesBetween) {
+            List<Employee> employeesListCopy = new ArrayList<>(employeeList);
+            List<Employee> unavailableEmployees = new ArrayList<>(extractRandomSubList(employeesListCopy, 0.3));
+            employeesListCopy.removeAll(unavailableEmployees);
+            List<Employee> undesiredEmployees = new ArrayList<>(extractRandomSubList(employeesListCopy, 0.3));
+            employeesListCopy.removeAll(undesiredEmployees);
+            List<Employee> desiredEmployees = new ArrayList<>(extractRandomSubList(employeesListCopy, 0.3));
+            employeesListCopy.removeAll(desiredEmployees);
+
+            unavailableEmployees.forEach((e) -> {
+                EmployeeAvailability toAdd = new EmployeeAvailability(tenantId, e, date, OffsetTime.of(LocalTime.MIN, config.getTimeZone().getRules().getOffset(date.atStartOfDay())), OffsetTime.of(LocalTime.MAX, config
+                        .getTimeZone().getRules().getOffset(date.atTime(LocalTime.MAX))));
+                toAdd.setState(EmployeeAvailabilityState.UNAVAILABLE);
```

Comment:
I am not seeing any randomness in here.
Ideally we want 80% stability and 20% randomness - just like in real life.

Let's merge this and talk about improving the generated data as a separate issue.